src/molecules/Modal/Modal.native.js

```diff
@@ -0,0 +1,246 @@
+  type,
+  visible,
+  onClose,
+  onBackdropClick,
+  showDragBar,
+  onSwipeComplete,
+  onBackButtonPress,
+}) => {
+  return type === 'centered' ? (
+    <Space margin={[3]}>
+      <RNModal
+        isVisible={visible}
+        onBackdropPress={onBackdropClick}
+        onBackButtonPress={onBackButtonPress}
+      >
+        <CenteredModalBody>
+          {onClose ? (
+            <Flex alignSelf="flex-end">
+              <Space padding={[0.5, 1, 0, 0]}>
+                <View>
+                  <Button
+                    testID="centered-cross"
+                    variant="tertiary"
+                    variantColor="tone" /**TODO: button uses shade of 800 but Figma is showing 980*/
+                    icon="close"
+                    size="medium"
+                    onClick={onClose}
+                  />
+                </View>
+              </Space>
+            </Flex>
+          ) : null}
+          {React.Children.map(children, (child) => {
+            return React.cloneElement(child, { ...child.props, type });
+          })}
+        </CenteredModalBody>
+      </RNModal>
+    </Space>
+  ) : type === 'fullscreen' ? (
+    <Space margin={[3, 0, 0, 0]}>
+      <RNModal isVisible={visible} hasBackdrop={false} onBackButtonPress={onBackButtonPress}>
+        <Flex flex={1}>
+          <FullscreenModalBody>
+            {React.Children.map(children, (child) => {
+              return React.cloneElement(child, { ...child.props, type, onClose });
+            })}
+          </FullscreenModalBody>
+        </Flex>
+      </RNModal>
+    </Space>
+  ) : type === 'bottomsheet' ? (
+    <Space margin={[0]}>
+      <Flex justifyContent="flex-end">
+        {/**TODO: confirm backdrop for bottomsheetModal */}
+        <RNModal
+          isVisible={visible}
+          swipeDirection={['down']}
+          onBackdropPress={onBackdropClick}
+          onSwipeComplete={onSwipeComplete}
+          onBackButtonPress={onBackButtonPress}
+        >
+          <BottomsheetModalBody>
+            {showDragBar ? (
+              <Flex alignItems="center">
+                <Space padding={[1, 0, 1.5, 0]}>
+                  <View>
+                    <Size height={0.5} width={8}>
+                      <FullScreenDragBar />
+                    </Size>
+                  </View>
+                </Space>
+              </Flex>
+            ) : null}
+            {React.Children.map(children, (child) => {
+              return React.cloneElement(child, { ...child.props, type });
+            })}
+          </BottomsheetModalBody>
+        </RNModal>
+      </Flex>
+    </Space>
+  ) : null;
+};
+
+Modal.Header = ModalHeader;
+Modal.Content = ModalContent;
+Modal.Footer = ModalFooter;
+
+Modal.propTypes = {
+  children: PropTypes.node.isRequired,
+  type: PropTypes.oneOf(['bottomsheet', 'centered', 'fullscreen']),
+  visible: PropTypes.bool,
+  onClose: PropTypes.func,
+  onBackdropClick: PropTypes.func,
+  onSwipeComplete: PropTypes.func,
```

Comment:
why do you need `onSwipeComplete` and  `onBackButtonPress` prop? Why can't it be same as `onClose`?

Reply:
`onSwipeComplete` and `onBackButtonPress` are the props exposed by `react-native-modal`. Do you mean to say that `onSwipeComplete` and `onSwipeComplete` would always close the modal?

Reply:
`onSwipeComplete` and `onBackButtonPress` are the props exposed by `react-native-modal`. Did you mean that `onSwipeComplete` and `onSwipeComplete` would always close the modal?

Reply:
I'm saying pass the same function you get in `onClose` to `onSwipeComplete`, `onBackButtonPress`. Just because we are using third party libs doesn't mean that we should expose their props to our consumers. Also the use case you're mentioning I don't see a probability of someone doing that behavior.

Reply:
I will remove it, my only concern was the use case like those.